### PR TITLE
Issue#5 -Mesgna -Woldeab

### PR DESCRIPTION
**Pull request by Mesgna616**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Issue#5 CRUD and Star API Created 

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: server/starAPI.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+// Getting all star from the database
+
+router.get("/stars", async (req, res) => {
+	try {
+		const result = await db.query("SELECT * FROM stars");
+		res.json(result.rows);
+	} catch (error) {
+		logger.error(error);
+		res.status(200).json(error);
```
Comment (MarkFarmiloe) [inline server/starAPI.js:14]: You shouldn't be returning status 200 when there has been an error.
In this case, it should only happen if the database is down or something like that, so 500 is probably the one to use.

File: server/starDatabase.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+
+CREATE TABLE stars (
+  id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+  name VARCHAR(255) NOT NULL,
+  description TEXT,
+  user_id SERIAL,
```
Comment (MarkFarmiloe) [inline server/starDatabase.sql:7]: SERIAL is used to ask PG to generate the value each time, which will be an INT value.
That's fine for the Primary Key, but not for Foreign Keys as we need to supply the value.
SO you need to change this to user_id INT

**Review by MarkFarmiloe — CHANGES_REQUESTED** [review]

**Review by TsioneTemesgenAbraham — APPROVED** [review]
Hey Mesgna. Great job!
You did the right thing by creating another file for the STAR API. It helped me understand the code better.

File: server/starAPI.js (hunk 1 above)
Comment (MarkFarmiloe) [reply server/starAPI.js:14]: Not a big enough issue to hold up the merge.

**Review by MarkFarmiloe — APPROVED** [review]

**Pull request merged**